ote_cli/ote_cli/tools/demo.py

```diff
@@ -0,0 +1,202 @@
+        action="store_true",
+        help="This option enables writing performance metrics on displayed frame. "
+        "These metrics take into account not only model inference time, but also "
+        "frame reading, pre-processing and post-processing.",
+    )
+
+    add_hyper_parameters_sub_parser(parser, hyper_parameters, modes=("INFERENCE",))
+
+    return parser.parse_args(), template, hyper_parameters
+
+
+def get_predictions(task, frame):
+    """
+    Returns list of predictions made by task on frame and time spent on doing prediction.
+    """
+
+    empty_annotation = AnnotationSceneEntity(
+        annotations=[], kind=AnnotationSceneKind.PREDICTION
+    )
+
+    item = DatasetItemEntity(
+        media=Image(frame),
+        annotation_scene=empty_annotation,
+    )
+
+    dataset = DatasetEntity(items=[item])
+
+    start_time = time.perf_counter()
+    predicted_validation_dataset = task.infer(
+        dataset,
+        InferenceParameters(is_evaluation=True),
+    )
+    elapsed_time = time.perf_counter() - start_time
+    item = predicted_validation_dataset[0]
+    return item.annotation_scene.annotations, elapsed_time
+
+
+def main():
+    """
+    Main function that is used for model evaluation.
+    """
+
+    # Dynamically create an argument parser based on override parameters.
+    args, template, hyper_parameters = parse_args()
+    # Get new values from user's input.
+    updated_hyper_parameters = gen_params_dict_from_args(args)
+    # Override overridden parameters by user's values.
+    override_parameters(updated_hyper_parameters, hyper_parameters)
+
+    hyper_parameters = create(hyper_parameters)
+
+    labels = [
+        LabelEntity(l, template.task_type, id=ID(i)) for i, l in enumerate(args.labels)
+    ]
+
+    # Get classes for Task, ConfigurableParameters and Dataset.
+    taks_class = get_impl_class(template.entrypoints.base)
+    environment = TaskEnvironment(
+        model=None,
+        hyper_parameters=hyper_parameters,
+        label_schema=LabelSchemaEntity.from_labels(labels),
+        model_template=template,
+    )
+
+    model = ModelEntity(
+        configuration=environment.get_model_configuration(),
+        model_adapters={
+            "weights.pth": ModelAdapter(load_model_weights(args.load_weights))
+        },
+        train_dataset=None,
+    )
+    environment.model = model
+
+    task = taks_class(task_environment=environment)
+
+    capture = open_images_capture(args.input, args.loop)
+
+    elapsed_times = []
+    frame_index = 0
+    while True:
+        frame = capture.read()
+        if frame is None:
+            break
+
+        predictions, elapsed_time = get_predictions(task, frame)
+        elapsed_times.append(elapsed_time)
+        if len(elapsed_times) > 10:
+            elapsed_times = elapsed_times[1:]
+        elapsed_time = np.mean(elapsed_times)
+
+        frame = draw_predictions(
+            template.task_type, predictions, frame, args.fit_to_size
+        )
```

Comment:
I guess visualization can be places in the following `if` clause.

Reply:
fixed